src/xwayland/XWM.cpp

```diff
@@ -172,8 +172,10 @@ void CXWM::readProp(SP<CXWaylandSurface> XSURF, uint32_t atom, xcb_get_property_
             XSURF->state.appid.pop_back();
         XSURF->events.metadataChanged.emit();
     } else if (atom == XCB_ATOM_WM_NAME || atom == HYPRATOMS["_NET_WM_NAME"]) {
-        size_t len         = xcb_get_property_value_length(reply);
-        char*  string      = (char*)xcb_get_property_value(reply);
+        size_t len    = xcb_get_property_value_length(reply);
+        char*  string = (char*)xcb_get_property_value(reply);
+        if (reply->type != HYPRATOMS["UTF8_STRING"])
```

Comment:
what about `TEXT`?

Reply:
also added XCB_ATOM_STRING cause it was showing up in logs and was still utf8